### PR TITLE
Fix issue with detecting mobile to render appropriately on export popup.

### PR DESCRIPTION
**Pull request by ruchijn**
Fix: #1170 

Chrome: 

https://user-images.githubusercontent.com/18288628/120073547-ca68e000-c066-11eb-925e-2bdd0e981793.mp4

Safari:

https://user-images.githubusercontent.com/18288628/120074061-0f8e1180-c069-11eb-92dc-3a5a5eec942b.mp4

### CODE REVIEW

File: src/components/ModalExport.tsx [hunk 1]
```diff
@@ -61,7 +62,8 @@ const ModalExport = () => {
     filterFunction: !shouldIncludeMetaAttributes ? child => !isFunction(child.value) : undefined
   }).length
 
-  const exportWord = 'share' in navigator ? 'Share' : 'Download'
+  const { isMobile } = DetectUserDevice()
+  const exportWord = isMobile ? 'Share' : 'Download'
```
Comment (raineorshine) [inline src/components/ModalExport.tsx:66]: I would prefer a simpler touch detection mechanism. You can use [isTouch](https://github.com/cybersemics/em/blob/dev/src/browser.ts#L6) which is already defined.

**Review by raineorshine — CHANGES_REQUESTED** [review]

File: src/hooks/detectUserDevice.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react'
+
+/** Custom hook detectUserDevice.ts to detect user device. */
+const DetectUserDevice = () => {
```
Comment (raineorshine) [inline src/hooks/detectUserDevice.ts:4]: For future reference, hooks should always start with `use`, e.g. `useDetectUserDevice`.

File: src/components/ModalExport.tsx (hunk 1 above)
Comment (ruchijn) [reply src/components/ModalExport.tsx:66]: I saw existing `isTouch`. But won't it also be true for touch screen laptops? 🤔

Comment (raineorshine) [reply src/components/ModalExport.tsx]: Yes, but I don't want to risk a false negative on mobile. It's a reasonable approach though.

**Pull request merged**